src/analysis/transcendental/liouville_constant.lean

```diff
@@ -0,0 +1,161 @@
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import analysis.transcendental.liouville
+import data.nat.factorial
+
+/-!
+This files contains a construction of a family of Liouville number
+-/
+
+namespace is_liouville
+
+noncomputable theory
+open_locale nat big_operators
+open set real finset
+
+variable (m : ℕ)
+
+local notation `α` := ∑' (i : ℕ), 1 / (m : ℝ) ^ i!
+local notation `α_first` k `terms` :=  ∑ i in range (k+1), 1 / (m : ℝ) ^ i!
+local notation `α_terms_after` k := ∑' i, 1 / (m : ℝ) ^ (i + (k+1))!
+
+lemma summable_inv_pow_fact {hm : 2 < m} : summable (λ i, 1 / (m : ℝ) ^ i!) :=
+begin
+  apply @summable_of_nonneg_of_le _ (λ n, (1 / m : ℝ) ^ n),
+  { intro b, rw one_div_nonneg, apply pow_nonneg, norm_num },
+  { intro b, dsimp only [],
+    rw [one_div, div_pow, one_pow, one_div, inv_le_inv],
+    { apply pow_le_pow _ (nat.self_le_factorial b),
+      norm_num, linarith },
+    repeat { apply pow_pos, norm_num, linarith } },
+  { apply summable_geometric_of_abs_lt_1, rw abs_of_nonneg,
+    { rw div_lt_one (by {norm_num, linarith} : 0 < (m : ℝ)), norm_num, linarith },
+    { rw one_div_nonneg, norm_num } }
+end
+
+lemma summable_inv_pow_n_add_fact {hm : 2 < m} (n : ℕ) :
+  summable (λ i, 1 / (m : ℝ) ^ (i + (n + 1))!) :=
+begin
+  apply @summable_of_nonneg_of_le _ (λ n, (1 / m : ℝ) ^ n),
+  { intro b, rw one_div_nonneg, apply pow_nonneg, norm_num },
+  { intro b, dsimp only [],
+    rw [one_div, div_pow, one_pow, one_div, inv_le_inv],
+    apply pow_le_pow,
+    repeat { apply pow_pos <|> norm_num <|> linarith },
+    transitivity b!,
+    { exact nat.self_le_factorial _ },
+    { exact nat.factorial_le (nat.le.intro rfl) }},
+  { apply summable_geometric_of_abs_lt_1, rw abs_of_nonneg,
+    { rw div_lt_one (by {norm_num, linarith} : 0 < (m : ℝ)), norm_num, linarith },
+    { rw one_div_nonneg, norm_num } }
+end
+
+lemma rat_of_α_k {hm : 2 < m} : ∀ k, ∃ p : ℕ, α_first k terms = p / ((m : ℝ) ^ k!) :=
+λ k, begin
+  induction k with k h,
+  { use 1,
+    simp only [sum_singleton, nat.cast_one, range_one] },
+  { rcases h with ⟨p_k, h_k⟩,
+    use p_k * (m ^ ((k + 1)! - k!)) + 1,
+    rw [sum_range_succ, h_k, div_add_div, div_eq_div_iff, one_mul, add_mul],
+    { norm_cast,
+      rw [add_mul, one_mul, nat.factorial_succ, show k.succ * k! - k! = (k.succ - 1) * k!,
+        by rw [nat.mul_sub_right_distrib, one_mul], nat.succ_sub_one, nat.succ_eq_add_one, add_mul,
+        one_mul, pow_add], ring },
+    rw mul_ne_zero_iff, split,
+    all_goals { refine pow_ne_zero _ _, norm_num, linarith } }
+end
+
+lemma α_terms_after_pos {hm : 2 < m} : ∀ k, 0 < α_terms_after k := λ n,
+calc 0 < 1 / (m : ℝ) ^ (n + 1)! : by { rw one_div_pos, apply pow_pos, norm_num, linarith }
+  ... = 1 / (m : ℝ) ^ (0 + (n + 1))! : by rw zero_add
+  ... ≤ ∑' (i : ℕ), 1 / (m : ℝ) ^ (i + (n + 1))! :
+      begin
+        refine le_tsum (@summable_inv_pow_n_add_fact _ hm _) 0 (λ _ _, _),
```

Comment:
The reason you need the `@` is because you are making `hm` an implicit argument `{hm : 2 < m}` in the lemmas upstairs. I think it should probably be explicit throughout this file.

Reply:
I have made them explicit